flutter/flutter
# Remove @objc inference build setting

## Description
- Remove SWIFT_SWIFT3_OBJC_INFERENCE build setting override in Swift app template.
- Remove stray DEVELOPMENT_TEAM from Swift and Objective-C app project template.
- Add CLANG_ENABLE_MODULES build setting override to Profile configuration to match Debug and Release in Swift app template.

To test:
```
$ flutter create -i swift test_swift
$ open ios/Runner.xcworkspace 
```
No @objc inference warning.
Set the environment variable SWIFT_DEBUG_IMPLICIT_OBJC_ENTRYPOINT to 3 in the Scheme > Run > Environment Variables.  Run the app and see no swift_objc_swift3ImplicitObjCEntrypoint crashes.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/34367.

## Tests

None.  
This is a build system warning, and I can't figure out how to get Xcode to treat those warnings as errors.  That may not even be desirable behavior since those warnings would be very dependent on the installed devtools version.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.


## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
